packages/dx-react-grid/src/plugins/search-state.test.jsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
+import { setupConsole } from '@devexpress/dx-testing';
+import { PluginHost } from '@devexpress/dx-react-core';
+import { changeSearchValue } from '@devexpress/dx-grid-core';
+import { pluginDepsToComponents, getComputedState, executeComputedAction } from './test-utils';
+import { SearchState } from './search-state';
+
+jest.mock('@devexpress/dx-grid-core', () => ({
+  changeSearchValue: jest.fn(),
+  pushSearchFilterExpr: jest.fn().mockImplementation(() => jest.fn().mockReturnValue('filters')),
+}));
+
+const defaultDeps = {
+  getter: {
+    rows: [{ id: 1 }],
+    columns: [{ name: 'a' }, { name: 'b' }],
+  },
+};
+
+describe('Search state', () => {
```

Comment:
I don't see a test with several action calls. And test for action in the controlled mode.